base/boot.jl

```diff
@@ -967,6 +968,7 @@ arraysize(a::Array, i::Int) = sle_int(i, nfields(a.size)) ? getfield(a.size, i)
 export arrayref, arrayset, arraysize, const_arrayref
 
 # For convenience
-EnterNode(old::EnterNode, new_dest::Int) = EnterNode(new_dest)
+EnterNode(old::EnterNode, new_dest::Int) = isdefined(old, :scope) ?
+    EnterNode(new_dest, old.scope) : EnterNode(new_dest)
```

Comment:
```suggestion
function EnterNode(node::EnterNode; 
                   dest::Int=node.dest,
                   scope)
    @nospecialize scope
    if !(@isdefined scope)
        isdefined(node, :scope) || return EnterNode(dest)
        scope = node.scope
    end
    return EnterNode(dest, scope)
end
```
Considering the possibility that `EnterNode` could get more complicated in the future (I'm not sure if it is likely though), defining it as `Effects(effects::Effects; overrides...)`-like constructor might be a better move?